### DISCUSSION TITLE
Revamp of the dev server & builder

### DISCUSSION THREAD

**Discussion by Zk2u** [flareact/flareact#181] [Ideas]
Yo!

I want to propose we switch from Webpack + `wrangler dev` to [esbuild](https://esbuild.github.io/) and [Miniflare](https://miniflare.dev/).

### Why?

Speed, mostly. Flareact's dev server is painfully slow when using `wrangler dev`. I've managed to switch to using Miniflare and it's significantly faster, however still painfully slow to develop on due to Webpack being slow.

Switching to Miniflare over `wrangler dev` was relatively easy, instead of `yarn dev` pointing at `flareact dev`, it runs `miniflare --watch` which speeds it up a bunch and has more features by running it locally.

Switching to esbuild over Webpack is probably going to be a lot harder and I haven't managed to do it yet. esbuild is the only thing really holding Flareact back from perhaps becoming faster than Next.js in both production and development.

<img width="854" alt="Screenshot 2021-09-26 at 11 57 04" src="https://user-images.githubusercontent.com/67651251/134804871-917bb028-7dc6-40d3-b2b4-64a9c0e780cf.png">

If we had 0.37s build times or less via esbuild instead of the 5-15s I have to wait for Webpack to fully rebuild everything, Flareact would be a beauty to develop on. The only thing holding it back right now (imo) is how slow DX is.

Not sure how we'd do it, but happy to help where I can. Have a think about it!

**Comment by jplhomer** [comment]
I think this is a terrific idea! I also don't know exactly how we'd do it 😄 happy to brainstorm here, though.

**Comment by tj-kev** [comment]
Also has an issue page over at #182 .
